parse.y

```diff
@@ -2587,7 +2587,81 @@ rb_parser_string_hash_cmp(rb_parser_string_t *str1, rb_parser_string_t *str2)
             memcmp(ptr1, ptr2, len1) != 0);
 }
 #endif
-#endif
+
+static void
+rb_parser_ary_extend(rb_parser_t *p, rb_parser_ary_t *ary, long len)
+{
+    long i;
+    if (ary->capa < len) {
+        ary->capa = len;
+        if (ary->data == NULL) {
+            ary->data = xcalloc(len, sizeof(void *));
+        } else {
+            ary->data = xrealloc(ary->data, sizeof(void *) * len);
+        }
+        for (i = ary->len; i < len; i++) {
+            ary->data[i] = 0;
+        }
+    }
+}
+
+static rb_parser_ary_t *
+rb_parser_ary_new_capa(rb_parser_t *p, long len)
+{
+    if (len < 0) {
+        rb_bug("negative array size (or size too big): %ld", len);
+    }
+    rb_parser_ary_t *ary = xcalloc(1, sizeof(rb_parser_ary_t));
+    long i;
+    ary->len = 0;
+    ary->capa = len;
+    if (0 < len) {
+        ary->data = xcalloc(len, sizeof(void *));
+    }
+    else {
+        ary->data = NULL;
+    }
+    for (i = 0; i < len; i++) {
+        ary->data[i] = 0;
+    }
```

Comment:
`xcalloc` zero-fills as well as the standard `calloc`.

Reply:
Thank you. Fixed